<a href="https://colab.research.google.com/github/adamarispadilla/final-project/blob/main/Padilla_3_16_2022_Lab_K_nearest_neighbors.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Using play_or_not.csv**

In [ ]:
import pandas as pd
import numpy as np
from sklearn import preprocessing


In [ ]:
df=pd.read_csv('play_or_not.csv')
df

weather  temp  play
0      sunny   Hot    No
1      sunny   Hot    No
2   overcast   Hot   Yes
3      rainy  Mild   Yes
4      rainy  Cool  Yes 
5      rainy  Cool    No
6   overcast  Cool   Yes
7      sunny  Mild    No
8      sunny  Cool   Yes
9      rainy  Mild   Yes
10     sunny  Mild   Yes
11  overcast  Mild   Yes
12  overcast   Hot   Yes
13     rainy  Mild    No

In [ ]:
#creating labelEncoder
le=preprocessing.LabelEncoder()

#converting string labels into numbers
weather_encoded=le.fit_transform(df['weather'])
print(weather_encoded)

[2 2 0 1 1 1 0 2 2 1 2 0 0 1]


In [ ]:
#converting string labels into numbers
temp_encoded=le.fit_transform(df['temp'])
print(temp_encoded)

label=le.fit_transform(df['play'])
print(label)

#combining weather and temp into single listof tuples
features=list(zip(weather_encoded, temp_encoded))
print(features)

[1 1 1 2 0 0 0 2 0 2 2 2 1 2]
[0 0 1 1 2 0 1 0 1 1 1 1 1 0]
[(2, 1), (2, 1), (0, 1), (1, 2), (1, 0), (1, 0), (0, 0), (2, 2), (2, 0), (1, 2), (2, 2), (0, 2), (0, 1), (1, 2)]


In [ ]:
#Let's build our KNN classifier model
from sklearn.neighbors import KNeighborsClassifier

model=KNeighborsClassifier(n_neighbors=3)

#Train the model using the training sets
model.fit(features,label)

#Predict Output
predicted=model.predict([[0,2]]) #0:Overcast, 2:Mild
print(predicted)

[1]


After gathering data regarding weather conditions, this algorithm will help us decide whether it is a play day ot not play day. After converting the strings to integers using the encoder, we get numbers ranging from 0 to 2. We plug in those numbers to the KNN classifier model and if the model returns a 0, it means a "no play" day and 1 is a "play" day. The model outputs a 1 which means it is a play day. 

In [ ]:
#import scikit-learn dataset library
from sklearn import datasets
import pandas as pd
import numpy as np

#load dataset
wine=datasets.load_wine()
wine

{'DESCR': '.. _wine_dataset:\n\nWine recognition dataset\n------------------------\n\n**Data Set Characteristics:**\n\n    :Number of Instances: 178 (50 in each of three classes)\n    :Number of Attributes: 13 numeric, predictive attributes and the class\n    :Attribute Information:\n \t\t- Alcohol\n \t\t- Malic acid\n \t\t- Ash\n\t\t- Alcalinity of ash  \n \t\t- Magnesium\n\t\t- Total phenols\n \t\t- Flavanoids\n \t\t- Nonflavanoid phenols\n \t\t- Proanthocyanins\n\t\t- Color intensity\n \t\t- Hue\n \t\t- OD280/OD315 of diluted wines\n \t\t- Proline\n\n    - class:\n            - class_0\n            - class_1\n            - class_2\n\t\t\n    :Summary Statistics:\n    \n    ============================= ==== ===== ======= =====\n                                   Min   Max   Mean     SD\n    ============================= ==== ===== ======= =====\n    Alcohol:                      11.0  14.8    13.0   0.8\n    Malic Acid:                   0.74  5.80    2.34  1.12\n    Ash:           

In [ ]:
print(wine.data.shape)

(178, 13)


In [ ]:
#Import train_test_split function
from sklearn.model_selection import train_test_split

#split dataset into training set and test set
#70% training and 30% test
X_train, X_test, y_train, y_test=train_test_split(wine.data, wine.target, test_size=0.3) 


In [ ]:
#Generating a model with K=3
#Import k-nearest neighbors classifier model
from sklearn.neighbors import KNeighborsClassifier

#Create KNN Classifier
knn=KNeighborsClassifier(n_neighbors=3)

#Train the model using the training sets
knn.fit(X_train,y_train)

#Predict the response for test dataset
y_pred=knn.predict(X_test)

In [ ]:
#Import scikit-learn metrics module for accuracy calculation
from sklearn import metrics

#Model accuracy, how often is the classifier correct?
print("Accuracy:", metrics.accuracy_score(y_test, y_pred))


Accuracy: 0.7592592592592593


This displays the accuracy of the algorithm we trained given the wine data. This means that given 3 neighbors, our model will be ~76% accurate.